apps/builder/app/builder/features/style-panel/sections/borders/border-property.tsx

```diff
@@ -0,0 +1,177 @@
+  borderPropertyOptions,
+  label,
+}: Pick<
+  RenderCategoryProps,
+  "currentStyle" | "setProperty" | "deleteProperty" | "createBatchUpdate"
+> & {
+  individualModeIcon?: ReactNode;
+  borderPropertyOptions: Partial<{
+    [property in StyleProperty]: { icon?: ReactNode };
+  }>;
+  label: string;
+}) => {
+  const borderProperties = Object.keys(borderPropertyOptions) as Array<
+    keyof typeof borderPropertyOptions
+  >;
+
+  const allPropertyValuesAreEqual =
+    new Set(
+      borderProperties.map(
+        (property) =>
+          toValue(currentStyle[property]?.value) ??
+          toValue(borderPropertyStyleValueDefault)
+      )
+    ).size === 1;
+
+  // If individualModeIcon is not defined, we do not want to display individual properties at all.
+  const [showIndividualMode, setShowIndividualMode] = useState(
+    () =>
+      allPropertyValuesAreEqual === false && individualModeIcon !== undefined
+  );
+
+  /**
+   * We do not use shorthand properties such as borderWidth or borderRadius in our code.
+   * However, in the UI, we can display a single field, and in that case, we can use any property
+   * from the shorthand property set and pass it instead.
+   **/
+  const firstPropertyName = borderProperties[0];
+
+  const { items: borderPropertyItems } = styleConfigByName[firstPropertyName];
+
+  const borderWidthKeywords = borderPropertyItems.map((item) => ({
+    type: "keyword" as const,
+    value: item.name,
+  }));
+
+  /**
+   * If the property is displayed in a non-individual mode, we need to provide a value for it.
+   * In Webflow, an empty value is shown. In Figma, the "Mixed" keyword is shown.
+   * We have decided to show the first defined value, as it is difficult to determine a maximum value
+   * when there are keywords (such as "thin" or "thick") and different units involved.
+   **/
+  const borderWidthStyleInfo = borderProperties
+    .map((property) => currentStyle[property]?.value)
+    .find((styleValue) => {
+      if (styleValue === undefined) {
+        return false;
+      }
+
+      return (
+        (styleValue.type === "unit" && styleValue.value > 0) ||
+        styleValue.type === "keyword"
+      );
+    });
+
+  const borderWidthStyleSource = getStyleSource(
+    ...borderProperties.map((property) => currentStyle[property])
+  );
+
+  const deleteAllProperties: DeleteProperty = (_propertyName, options) => {
+    const batch = createBatchUpdate();
+    for (const property of borderProperties) {
+      batch.deleteProperty(property);
+    }
+    batch.publish(options);
+  };
+
+  const setAllProperties: ReturnType<SetProperty> = (value, options) => {
+    const batch = createBatchUpdate();
+    for (const property of borderProperties) {
+      batch.setProperty(property)(value);
+    }
+    batch.publish(options);
+  };
+
+  return (
+    <Grid gap={1}>
+      <Grid
+        css={{
+          gridTemplateColumns: individualModeIcon
+            ? "1fr 80px max-content"
+            : "1fr 116px",
```

Comment:
what are those hardcoded values? I don't think we should have any, either design is wrong or you used wrong layout or something, it should almost always fit some spacing token